### ISSUE TITLE
Installing an apk file

### ISSUE THREAD

**Issue by bizzguy** [mttkay/replicant#18]
I would like to be able to install an apk from the project directory.  I'm using AndroidStudio and starting replicant from the module directly.  The apk file would be in "build/apk".  There also might be more than one apk.

**Comment by mttkay** [comment]
I'm not sure what the problem is? You install it as you would do outside replicant, just that it saves you a few keystrokes:

in replicant:
`> install path/to/apk`

are you saying this is not working for you?

**Comment by bizzguy** [comment]
What you're suggesting does work but I'd like replicant to do even more!  Maybe I'm just being greedy here :)

My path names and apk names tend to be long.  And I'm dealing with many projects.  

I'd like to be able to type "install" and have replicant figure out the correct path and APK name based on the current directory.  Also, since there may be more than one apk (i.e. debug and release) I would like it to default to debug unless I specify the flavor (i.e. "install release").

**Comment by mttkay** [comment]
Not at all, I was thinking about this in fact. I then abandoned the idea again because build output is both dependent on the build system used and the project itself. I do still like the idea though, but I wonder what the best way would be to to implement this without the feature trying to be too clever for its own good if you know what I mean.

Scanning the project folder for APKs is the easiest part. But as you say, figuring out the right one in case there's more than one is tricky. One option would be to simply ask the user which one they meant by presenting a list of APKs found. This, however, requires a new type of command, one which is able to wait for and respond to user input itself (right now, commands are simple carry-out tasks triggered by the REPL.)

I'll think about it some more; meanwhile let me know if you have ideas or preferences around how you'd like to see this working.

**Comment by bizzguy** [comment]
I'm working with AndroidStudio and the new build system so I'd like to see it search the "build/apk" directories for the APK files.  I suppose it could also search the "build.gradle" file for product flavor.

Having REPL be able to ask for user input might open up a whole new range of possibilities so I'd like to upvote that!
